packages/portfolio-contract/test/rebalance.test.ts

```diff
@@ -115,328 +146,358 @@ test('solver all to one (B + C -> A)', async t => {
-    brand: TOK_BRAND,
-    feeBrand: FEE_BRAND,
-  });
-  t.deepEqual(steps, [
-    { src: '<Deposit>', dest: '@agoric', amount: token(100n) },
-    // TODO dckc should this  go through +agoric?
-    // { src: '+agoric', dest: '@agoric', amount: token(100n) },
-    { src: '@agoric', dest: '@noble', amount: token(100n) },
-    {
-      src: '@noble',
-      dest: '@Arbitrum',
-      amount: token(70n),
-      fee: fee(15_000_000n),
-    },
-    {
-      src: '@noble',
-      dest: '@Avalanche',
-      amount: token(30n),
-      fee: fee(15_000_000n),
-    },
-    { src: '@Arbitrum', dest: A, amount: token(70n), fee: fee(15_000_000n) },
-    { src: '@Avalanche', dest: B, amount: token(30n), fee: fee(15_000_000n) },
-  ]);
-});
+testWithAllModes(
+  'solver deposit redistribution (Deposit 100 -> A 70, B 30)',
+  async (t, mode) => {
+    const current = balances({ '<Deposit>': 100n, [A]: 0n, [B]: 0n });
+    const { steps } = await planRebalanceFlow({
+      mode: mode as RebalanceMode,
+      network: TEST_NETWORK,
+      current,
+      target: { '<Deposit>': ZERO, [A]: token(70n), [B]: token(30n) },
+      brand: TOK_BRAND,
+      feeBrand: FEE_BRAND,
+    });
+    t.deepEqual(steps, [
+      { src: '<Deposit>', dest: '@agoric', amount: token(100n) },
+      // TODO dckc should this  go through +agoric?
+      // { src: '+agoric', dest: '@agoric', amount: token(100n) },
+      { src: '@agoric', dest: '@noble', amount: token(100n) },
+      {
+        src: '@noble',
+        dest: '@Arbitrum',
+        amount: token(70n),
+        fee: fee(15_000_000n),
+      },
+      {
+        src: '@noble',
+        dest: '@Avalanche',
+        amount: token(30n),
+        fee: fee(15_000_000n),
+      },
+      { src: '@Arbitrum', dest: A, amount: token(70n), fee: fee(15_000_000n) },
+      { src: '@Avalanche', dest: B, amount: token(30n), fee: fee(15_000_000n) },
+    ]);
+  },
+);
 
-test('solver withdraw to cash (A 50 + B 30 -> Cash)', async t => {
-  const current = balances({ [A]: 50n, [B]: 30n, '<Cash>': 0n });
-  const { steps } = await planRebalanceFlow({
-    mode: 'cheapest',
-    network: TEST_NETWORK,
-    current,
-    target: { [A]: ZERO, [B]: ZERO, '<Cash>': token(80n) },
-    brand: TOK_BRAND,
-    feeBrand: FEE_BRAND,
-  });
-  t.deepEqual(steps, [
-    { src: A, dest: '@Arbitrum', amount: token(50n), fee: fee(15_000_000n) },
-    { src: '@Arbitrum', dest: '@noble', amount: token(50n) },
-    { src: B, dest: '@Avalanche', amount: token(30n), fee: fee(15_000_000n) },
-    { src: '@Avalanche', dest: '@noble', amount: token(30n) },
-    { src: '@noble', dest: '@agoric', amount: token(80n) },
-    { src: '@agoric', dest: '<Cash>', amount: token(80n) },
-  ]);
-});
+testWithAllModes(
+  'solver withdraw to cash (A 50 + B 30 -> Cash)',
+  async (t, mode) => {
+    const current = balances({ [A]: 50n, [B]: 30n, '<Cash>': 0n });
+    const { steps } = await planRebalanceFlow({
+      mode: mode as RebalanceMode,
```

Comment:
why isn't mode already correctly typed?

Reply:
No idea. It's test code and I didn't want to waste more time on it.